test/rulesets/Core/core.jl

```diff
@@ -0,0 +1,12 @@
+@testset "typeassert" begin
+    test_rrule(typeassert, 1.0, Float64 ⊢ NoTangent())
+    test_frule(typeassert, 1.0, Float64 ⊢ NoTangent())
+end
+
+@testset "ifelse" begin
+    test_rrule(ifelse, true ⊢ NoTangent(), 1.0, 2.0)
+    test_frule(ifelse, false ⊢ NoTangent(), 1.0, 2.0)
+
+    test_rrule(ifelse, true ⊢ NoTangent(), [1.0], [2.0]; check_inferred=false)
+    test_frule(ifelse, false ⊢ NoTangent(), [1.0], [2.0]; check_inferred=false)
```

Comment:
```suggestion
    test_rrule(ifelse, true, 1.1, 2.0)
    test_frule(ifelse, false, 1.1, 2.0)

    test_rrule(ifelse, true, [1.1], [2.0]; check_inferred=false)
    test_frule(ifelse, false, [1.1], [2.0]; check_inferred=false)
```
`true`/`false` automatically gets a `NoTangent()`. And `1.0`s and `0.0`s are sometimes tricky inputs as they spuriously give the right answer